lib/mwUtil.js

```diff
@@ -412,13 +412,21 @@ mwUtil.isSelfRedirect = (req, location) => {
 
 /**
  * Fetches the summary content from the provided URI
- * @param  {HyperSwitch}  hyper HyperSwitch context
- * @param  {URI}          uri   summary URI
- * @return {string}             summary body content
+ * @param  {HyperSwitch}  hyper     HyperSwitch context
+ * @param  {URI}          uri       summary URI
+ * @param  {headers}      headers   req header
+ * @param  {headers}      pRes      response of the parent request
+ * @return {string}                 summary body content
  */
-mwUtil.fetchSummary = (hyper, uri) => {
-    return hyper.get({ uri })
+mwUtil.fetchSummary = (hyper, uri, headers, pRes) => {
+    return hyper.get({ uri, headers })
     .then((res) => {
+        // Assign content-language and vary header to parent response
+        // based on one of summary responses
+        if (pRes && pRes.headers && !pRes.headers['content-language']) {
+            pRes.headers['content-language'] = res.headers && res.headers['content-language'];
+            mwUtil.addVaryHeader(pRes, 'accept-language');
+        }
```

Comment:
Maybe we can return an object [summary, content-language] instead of assigning the header here. I think it would be less magical.